### PR TITLE
General refactor and cleanup of Diagnostics Asp.Net and Core Error Reporting integration tests and snippets.

### PR DESCRIPTION
**Pull request by amanda-tarafa**
This PR:

- Extracted the ErrorEvent verifiers from AspNet.IntegrationTests to Common.IntegrationTests
- Makes AspNetCore.IntegrationTests and Snippets use the ErrorEvent verifiers extracted above.
- General cleanup and refactoring of ErrorReporting Integration tests and Snippets.

### CODE REVIEW

File: apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common.IntegrationTests/ErrorReporting/ErrorEventEntryVerifiers.cs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+using System.Net.Http;
+using Xunit;
+
+namespace Google.Cloud.Diagnostics.Common.IntegrationTests.ErrorReporting
+{
+    public static class ErrorEventEntryVerifiers
+    {
+        /// <summary>
+        /// Checks that an <see cref="ErrorEvent"/> contains valid data,
+        /// including HTPP Context data.
```
Comment (iantalarico) [inline apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common.IntegrationTests/ErrorReporting/ErrorEventEntryVerifiers.cs]: nit: HTTP

File: apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common.IntegrationTests/ErrorReporting/ErrorEventEntryVerifiers.cs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        public static void VerifyFullErrorEventLogged(ErrorEvent errorEvent, string testId, string functionName, int responseStatusCode)
+        {
+            VerifyErrorEventLogged(errorEvent, testId, functionName);
+            VerifyHttpContextLogged(errorEvent, responseStatusCode);
+        }
+
+        /// <summary>
+        /// Checks that an <see cref="ErrorEvent"/> contains valid data.
+        /// Doesn't check for HTTP Context data, which won't be logged 
+        /// when from withing a self hosted Web API application the
```
Comment (iantalarico) [inline apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common.IntegrationTests/ErrorReporting/ErrorEventEntryVerifiers.cs]: nit: within

File: apis/Google.Cloud.Diagnostics.AspNet/Google.Cloud.Diagnostics.AspNet.IntegrationTests/ErrorReporting/ErrorReportingTest.cs [hunk 3]
```diff
@@ -29,178 +29,136 @@
...
+            response = await requestTask3;
+            Assert.Equal(HttpStatusCode.InternalServerError, response.StatusCode);
+            response = await requestTask4;
+            Assert.Equal(HttpStatusCode.InternalServerError, response.StatusCode);
+
+            var errorEvents = s_polling.GetEvents(_startTime, _testId, 4);
+            Assert.Equal(4, errorEvents.Count());
+
+            var exceptionEvents = errorEvents
+            .Where(e => e.Message.Contains(nameof(ErrorReportingController.ThrowsException)));
```
Comment (iantalarico) [inline apis/Google.Cloud.Diagnostics.AspNet/Google.Cloud.Diagnostics.AspNet.IntegrationTests/ErrorReporting/ErrorReportingTest.cs]: nit: indent

File: apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common.IntegrationTests/ErrorReporting/ErrorEventEntryVerifiers.cs (hunk 1 above)
Comment (amanda-tarafa) [reply apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common.IntegrationTests/ErrorReporting/ErrorEventEntryVerifiers.cs]: Done :)

File: apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common.IntegrationTests/ErrorReporting/ErrorEventEntryVerifiers.cs (hunk 2 above)
Comment (amanda-tarafa) [reply apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common.IntegrationTests/ErrorReporting/ErrorEventEntryVerifiers.cs]: Done.

File: apis/Google.Cloud.Diagnostics.AspNet/Google.Cloud.Diagnostics.AspNet.IntegrationTests/ErrorReporting/ErrorReportingTest.cs (hunk 3 above)
Comment (amanda-tarafa) [reply apis/Google.Cloud.Diagnostics.AspNet/Google.Cloud.Diagnostics.AspNet.IntegrationTests/ErrorReporting/ErrorReportingTest.cs]: Done.

**Pull request merged**